File: platform-core-service/platform-core-service.Common/Models/DTOs/EntityDTO/Report/SelectReportDTO.cs
using platform_core_service.Common.Entities.BaseEntity;
using platform_core_service.Common.Utils.Enums;

namespace platform_core_service.Common.Models.DTOs.EntityDTO.Report
{
    public class SelectReportDTO : IBaseKey<string>
    {
        public string Id { get; set; } = null!;
        public ReportTargetType TargetType { get; set; }
        public string TargetId { get; set; } = null!;
        public ReportReason Reason { get; set; }
        public ReportStatus Status { get; set; }
        public DateTimeOffset? DateCreated { get; set; }
    }
}

File: frontend/src/types/report/select-report-dto.ts
import { ReportReason } from "./report-reason";
import { ReportStatus } from "./report-status";
import { ReportTargetType } from "./report-target-type";

export interface SelectReportDTO {
  id: string;
  targetType: ReportTargetType;
  targetId: string;
  reason: ReportReason;
  status: ReportStatus;
  dateCreated?: string | null;
}
